Library/OcMachoLib/CxxSymbolsX.h

```diff
@@ -0,0 +1,114 @@
+/** @file
+  Provides services for C++ symbols.
+
+Copyright (c) 2018, Download-Fritz.  All rights reserved.<BR>
+This program and the accompanying materials are licensed and made available
+under the terms and conditions of the BSD License which accompanies this
+distribution.  The full text of the license may be found at
+http://opensource.org/licenses/bsd-license.php.
+
+THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+**/
+
+#include "MachoX.h"
+
+MACH_NLIST_X *
+MACH_X (MachoGetMetaclassSymbolFromSmcpSymbol) (
+  IN OUT OC_MACHO_CONTEXT     *Context,
+  IN     CONST MACH_NLIST_X   *Smcp
+  )
+{
+  MACH_NLIST_X  *Symbol;
+  BOOLEAN       Result;
+
+  ASSERT (Context != NULL);
+  ASSERT (Smcp != NULL);
+  MACH_ASSERT_X;
```

Comment:
I might actually request `MACH_ASSERT_X` to be a functional macro taking `Context` as an argument, because relying on `Context` being the Mach-O context is very dirty and counter-intuitive.